config/params/config_utils_prod.go

```diff
@@ -7,26 +7,24 @@ import (
 	"github.com/mohae/deepcopy"
 )
 
-var beaconConfig = MainnetConfig()
-
 // BeaconConfig retrieves beacon chain config.
 func BeaconConfig() *BeaconChainConfig {
-	return beaconConfig
+	return Registry.GetActive()
 }
 
 // OverrideBeaconConfig by replacing the config. The preferred pattern is to
 // call BeaconConfig(), change the specific parameters, and then call
 // OverrideBeaconConfig(c). Any subsequent calls to params.BeaconConfig() will
 // return this new configuration.
 func OverrideBeaconConfig(c *BeaconChainConfig) {
-	beaconConfig = c
+	Registry.active = c
 }
 
 // Copy returns a copy of the config object.
 func (b *BeaconChainConfig) Copy() *BeaconChainConfig {
 	config, ok := deepcopy.Copy(*b).(BeaconChainConfig)
 	if !ok {
-		config = *beaconConfig
+		panic("somehow deepcopy produced a BeaconChainConfig that is not of the same type as the original")
```

Comment:
see note in the `config_utils_develop.go` version of this. Note this version is typically compiled for release, whereas the other file has a build tag that is only compiled in test/staging.